### DISCUSSION TITLE
Correct way to allow scripts to import other scripts?

### DISCUSSION THREAD

**Discussion by Hi-ImKyle** [sebastienros/jint#1659] [Q&A] [answered]
Currently I am just trying to import a basic math script that has a add function.

```js
// file: math.js
const add = (a, b) => { return a + b }
module.exports = {
    add
}
```

then in another file I am trying to import it. _each way of importing it was ran independently_*

```js
const math = require("./math.js")
print(math.add(4, 6))
// result: Jint.Runtime.JavaScriptException: require is not defined

import { add } from "./math.js"
print(add(4, 6))
// result: Esprima.ParserException: Line 4: Unexpected token

const math = import("./math.js")
print(math.add(4, 6))
// result: System.UriFormatException: Invalid URI: The URI is empty.
```

Forgive me if this has been asked before but I did look and couldn't find anything that best fit [nor did I find any real answers](https://github.com/sebastienros/jint/discussions/840)

**Comment by lahma** [comment]
I'd generally guide towards [the sample in main readme](https://github.com/sebastienros/jint#using-modules) and [the test suite showing different ways to handle module code](https://github.com/sebastienros/jint/blob/main/Jint.Tests/Runtime/ModuleTests.cs).

**Reply by Hi-ImKyle** [reply]
I did look at both. Am I wrong to be using `engine.Evaluate` then? Should I be using `AddModule/ImportModule`? How does that work with loading a file, if say I wanted to load `index.js` and then it goes and imports `math.js`?

**Reply by lahma** [reply, marked as answer]
Generally you need to be in module context when importing modules, Evaluate and Execute are not. 

If you've enabled modules with base path importing relative should work. You can override settings via engine's module options which allows also customizing loading logic.
